core/capabilities/targets/write_target.go

```diff
@@ -29,24 +31,39 @@ type WriteTarget struct {
 	forwarderAddress string
 	capabilities.CapabilityInfo
 	lggr logger.Logger
+
+	// options for configuring local node info retrieval
+	registry         core.CapabilitiesRegistry
+	localNodeRetryMs int // how often to retry fetching local node info
+	localNode        *capabilities.Node
 }
 
-func NewWriteTarget(lggr logger.Logger, id string, cr commontypes.ContractReader, cw commontypes.ChainWriter, forwarderAddress string) *WriteTarget {
+func NewWriteTarget(lggr logger.Logger, id string, cr commontypes.ContractReader, cw commontypes.ChainWriter, forwarderAddress string, registry core.CapabilitiesRegistry, localNodeInfoIntervalMsOpt ...int) *WriteTarget {
```

Comment:
Using splat notation to denote an optional variable in this case is a little odd -- I'd recommend refactoring this to a functional options pattern (`...func(c *Cfg)`)  so that other variables could also be set in this way, or just passing in an *int pointer, where you can check nil to fallback to the default.

Reply:
yeah, originally it was a nillable but it felt odd to have to always pass in nil. I'll revert back to that anyway since it seems overkill to do functional options pattern for a single option